# Project 1: Fast Food Marketing Campaign A/B Test


## Introduction
This project evaluates three marketing campaigns to determine which one generates the highest sales.

In [12]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

In [13]:
import os
print(os.listdir())

['Bonus 1.ipynb', 'Ex (3).ipynb', 'Ex(2).ipynb', 'Ex(4).ipynb', 'Ex(5).ipynb', 'Ex(6).ipynb', 'Ex(7).ipynb', 'Ex(8).ipynb', 'Ex(9).ipynb', 'Ex1.ipynb', 'm3-1_ex.ipynb', 'm3-2_ex.ipynb', 'm4-1.ipynb', 'Project 1.ipynb', 'WA_Marketing-Campaign.csv']


In [14]:
import pandas as pd

In [15]:
df = pd.read_csv("WA_Marketing-Campaign.csv")
df.head()

,MarketID,MarketSize,LocationID,AgeOfStore,Promotion,week,SalesInThousands
0,1,Medium,1,4,3,1,33.73
1,1,Medium,1,4,3,2,35.67
2,1,Medium,1,4,3,3,29.03
3,1,Medium,1,4,3,4,39.25
4,1,Medium,2,5,2,1,27.81


In [16]:
df.groupby("Promotion")["SalesInThousands"].mean()

Promotion
1    58.099012
2    47.329415
3    55.364468
Name: SalesInThousands, dtype: float64

## Results

The average sales for each promotion are as follows:

- Promotion 1: 58.09
- Promotion 2: 47.33
- Promotion 3: 55.36

Promotion 1 achieved the highest average sales among all campaigns.

In [17]:
group1 = df[df["Promotion"] == 1]["SalesInThousands"]
group2 = df[df["Promotion"] == 2]["SalesInThousands"]
group3 = df[df["Promotion"] == 3]["SalesInThousands"]

from scipy import stats
stats.f_oneway(group1, group2, group3)

F_onewayResult(statistic=np.float64(21.953485793080674), pvalue=np.float64(6.765849261408834e-10))

## Statistical Test (ANOVA)

The ANOVA test results show a p-value of 6.76e-10, which is less than 0.05.

This indicates that there is a statistically significant difference in sales between the three promotions.

In [18]:
stats.f_oneway(group1, group2, group3)

F_onewayResult(statistic=np.float64(21.953485793080674), pvalue=np.float64(6.765849261408834e-10))

In [19]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd

tukey = pairwise_tukeyhsd(
    endog=df["SalesInThousands"],
    groups=df["Promotion"],
    alpha=0.05
)

print(tukey)

 Multiple Comparison of Means - Tukey HSD, FWER=0.05 
group1 group2 meandiff p-adj   lower    upper  reject
-----------------------------------------------------
     1      2 -10.7696    0.0 -14.7738 -6.7654   True
     1      3  -2.7345 0.2444  -6.7388  1.2697  False
     2      3   8.0351    0.0   4.1208 11.9493   True
-----------------------------------------------------


### Post-hoc Test (Tukey HSD)

The Tukey HSD test results show the following:

- There is a significant difference between Promotion 1 and Promotion 2.
- There is NO significant difference between Promotion 1 and Promotion 3.
- There is a significant difference between Promotion 2 and Promotion 3.

These results indicate that Promotion 2 performs significantly worse than the other promotions, while Promotion 1 and Promotion 3 perform similarly.

Since Promotion 1 has the highest average sales, it is considered the best marketing strategy.

## Final Conclusion

Based on both descriptive analysis and statistical testing, Promotion 1 achieved the highest average sales.

The ANOVA test confirmed that the differences between promotions are statistically significant.

Therefore, Promotion 1 is the most effective marketing strategy and is recommended for implementation.